src/io/record-play-movie.cpp

```diff
@@ -167,56 +193,87 @@ static bool string_is_repeat(char *str, int len)
     return true;
 }
 
-static errr send_text_to_chuukei_server(TERM_LEN x, TERM_LEN y, int len, TERM_COLOR col, concptr str)
+#ifdef JP
+/* マルチバイト文字を分割せずに str を分割する長さを len 以下で返す */
+static int find_split(concptr str, int len)
 {
-    char buf[1024 + 32];
-    char buf2[1024];
-
-    strncpy(buf2, str, len);
-    buf2[len] = '\0';
+    /*
+     * SJIS が定義されている場合でも、str は常に EUC-JP としてエンコードされます。
+     * 他の場所で行われているような 3 バイトのエンコーディングは想定していません。
+     */
+    int last_split = 0, i = 0;
+    while (i < len) {
+        if (iseuckanji(str[i])) {
+            if (i < len - 1) {
+                last_split = i + 2;
+            }
+            i += 2;
+        } else {
+            last_split = i + 1;
+            ++i;
+        }
+    }
+    return last_split;
+}
+#endif
 
+static errr send_text_to_chuukei_server(TERM_LEN x, TERM_LEN y, int len, TERM_COLOR col, concptr str)
+{
```

Comment:
127 may become a magic number.
Please declare the variable (constexpr auto).
I have no comment more.

Reply:
The latest forced push adds a file-level static, split_max, for that constant and uses it in send_text_to_chuukei_server() and send_wipe_to_chuukei_server().